### ISSUE TITLE
BUG: projecta lister compound filtering and grouping revisited

### ISSUE THREAD

**Issue by sbillinge** [regro/regolith#481]
related to #480 and #482
desired behavior:
``rh l_projecta -f status finished -l mwinitch`` returns the pra that Michael has finished
current behavior:
seems to return all finished projecta, regardless of Michael or not

desired behavior:
``rh l_projecta -f status finished --group_by_lead`` returns all finished pra grouped by lead
current behavior:
returns the same as ``rh l_projecta -f status finished``

**Comment by jiaweizang** [comment]
I will take this.

**Comment by jiaweizang** [comment]
@sbillinge Hi Simon. I found that in l_projecta, we can currently use the filter function efficiently by --fliter. For example, 
"regolith helper l_projecta -f grants a status b lead c"
will return projecta whose grants is a, status is b, lead is c.
Specifically,
"regolith helper l_projecta -f grants dmref19 status finished lead mwinitch"
will return projecta whose grants is dmref19, status is finished, lead is mwinitch.

This function is done by search_collection and key_value_pair_filter in tools.py.

If we use --filter, we can delete many arguments, like --lead, --stati, --ended, --grant. And then a desired behavior and logic is like, first use --filter to filter the collection, and then the user decide if use --grp_by_lead, and then decide if use --verbose.
what do you think about?

**Comment by sbillinge** [comment]
yes, that's right.  It is the direction I want to move in because it is more abstract so we need fewer helpers but have all the functionality.  If we want to then have less typing for people for very common tasks, we can build lightweight helpers that use these functions in a particular way to get the job done. 

One piece of functionality that doesn't work in the current implementation of this I think is if there are nested dicts, so maybe you can think about that, e.g., ``-f name simon`` but ``-f employment current`` is something I might want to search for, but can't get not.  This would be the topic of a different issue/prum

**Comment by jiaweizang** [comment]
@mwinitch
